src/live-programming.rst

```diff
@@ -16,6 +16,379 @@ REPLs
+
+
+.. container:: bib-item
+
+  .. bibliography:: data-analysis-environments.bib
+    :filter: key == 'muller:2019'
+
+  This paper describes the practice of data science with a focus on how data science workers engage with data. It reports five approaches to data: data as given; as captured; as curated; as designed; and as created.
+
+This chapter reflects on the challenges in the common practice of data analysis, and reports novel designs that address the challenges.
+The rest of the chapter is organized as the following:
+1) the first part of the article introduces novel tools and systems that assist with data collection, cleaning, and analyzing,
+2) the second part of the article introduces computational notebooks, a common tool for documenting and sharing the process of data science.
+
+Tools for Data Collection
+--------------------------
+Data can be collected from a variety of sources.
+For example, data can be downloaded as a public dataset from online repositories, or generated by physical sensors (weather data, personal health data), or collected from existing websites (flight prices, stock prices), or manually crafted (the grade of students' assignments).
+In data science education, there is an increasing call to encourage students to collect and explore authentic data that are meaningful to themselves rather than providing them a standard dataset.
+The common practice for data science workers to collect data from an existing website is using web scraping tools.
+However, the additional effort to learn the scraping process may block novices' interests to explore the data.
+
+Web Scraping by Demonstration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+One approach that has been explored is using Programming by Demonstration (PBD) to facilitate the process of web data collection.
+Chasin et al. first implemented Helena_, a high-level programming language for automating repetitive interactions on webpages. 
+Chasin et al. then applied Programming by Demonstration (PBD) so that users only need to demonstrate how to collect data in the first row from a hierarchical website, and the system learns from the demonstration and generalizes to the other rows :cite:`chasins:2018`.
+They found that this novel approach could help users finish scraping tasks 8 times faster than a traditional web scraping tool (Selenium) through a within-subject study with 15 computer scientists.
+
+.. _Helena: http://helena-lang.org/
+
+.. container:: bib-item
+
+  .. bibliography:: data-analysis-environments2.bib
+    :filter: key == 'chasins:2018'
+    
+  This paper presents Rousillon, a programming-by-demonstration system for scraping hierarchically-structured data from different webpages. 
+
+Example-Centric Live Data Analysis Environments
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Another approach is to turn any webpage into a data analysis environment.
+**DS.js** :cite:`zhang:2017` explored this idea by introducing the idea of example-centric live programming.
```

Comment:
These should also be in the corresponding paper summary. Text outside a paper summary should be introducing a topic not focusing on the details of a specific paper.